packages/nextjs-demo/src/app/tts/api/route.tsx

```diff
@@ -0,0 +1,147 @@
+        }
+        controller.enqueue(value);
+        read();
+      }
+      read();
+    },
+  });
+  return new NextResponse(nextStream, { headers, status });
+}
+
+function makeSsml(voice: string, rate: number, text: string) {
+  return `
+  <speak version="1.0" xml:lang="en-US">
+    <voice xml:lang="en-US" name="${voice}">
+      <prosody rate="${Math.round((rate - 1.0) * 100)}%">${text}</prosody>
+    </voice>
+  </speak>`;
+}
+
+/**
+ * REST client for Eleven Labs TTS.
+ */
+function ttsEleven(voiceId: string, rate: number, text: string): Promise<Response> {
+  const latencyMode = 22;
+  const apiKey: string = process.env.ELEVEN_API_KEY ?? '';
+  const url: string = `https://api.elevenlabs.io/v1/text-to-speech/${voiceId}/stream?optimize_streaming_latency=${latencyMode}`;
+  const headers: HeadersInit = {
+    Accept: 'audio/mpeg',
+    'xi-api-key': apiKey,
+    'Content-Type': 'application/json',
+  };
+  const body: string = JSON.stringify({
+    text,
+    model_id: 'eleven_monolingual_v1',
+    voice_settings: {
+      stability: 0.5,
+      similarity_boost: 0.5,
+    },
+  });
+  return fetch(url, {
+    method: 'POST',
+    headers,
+    body,
+  });
+}
+
+/**
+ * REST client for Azure TTS.
+ */
+function ttsAzure(voice: string, rate: number, text: string): Promise<Response> {
+  const region = 'westus';
+  const apiKey = process.env.AZURE_TTS_API_KEY ?? '';
+  const outputFormat = 'audio-24khz-48kbitrate-mono-mp3';
+  const url = `https://${region}.tts.speech.microsoft.com/cognitiveservices/v1`;
+  const headers = new Headers();
+  headers.append('Ocp-Apim-Subscription-Key', apiKey);
+  headers.append('Content-Type', 'application/ssml+xml');
+  headers.append('X-Microsoft-OutputFormat', outputFormat);
+  headers.append('User-Agent', 'MyTTS');
+  return fetch(url, {
+    method: 'POST',
+    headers,
+    body: makeSsml(voice, rate, text),
+  });
+}
+
+/**
+ * REST client for AWS Polly TTS.
+ */
+function ttsAws(voice: string, rate: number, text: string) {
+  const region = 'us-west-2';
+  const outputFormat = 'mp3';
+  const queryParams = new URLSearchParams({
+    Text: text,
+    OutputFormat: outputFormat,
+    VoiceId: voice,
+  });
+  const url = `https://polly.${region}.amazonaws.com/v1/speech?${queryParams.toString()}`;
+  const headers = new Headers();
+  return fetch(url, { headers });
+}
+
+/**
+ * Returns a temporary API key for use in a WebSocket connection to the given provider.
+ * Currently, this is only configured for Eleven Labs, and even then, we're mostly
+ * faking it because Eleven doesn't support temporary API keys yet.
+ */
+export async function POST(request: NextRequest) {
+  const inJson = await request.json();
+  const provider = inJson.provider as string;
+  if (provider != 'eleven') {
+    return new NextResponse(JSON.stringify({ error: 'unknown provider' }));
+  }
+
+  const func = () => getApiKey('ELEVEN_API_KEY');
+  return new NextResponse(JSON.stringify({ token: await func() }));
```

Comment:
```suggestion
  return new NextResponse(JSON.stringify({ token: await getApiKey('ELEVEN_API_KEY') }));
```

Why won't this work?

Reply:
it will, just preparing for a future where we'll need to obtain API keys for different providers